### ISSUE TITLE
Support native |> pipes 

### ISSUE THREAD

**Issue by nischalshrestha** [nischalshrestha/Unravel#106]

**Comment by andreweatherman** [comment]
Any update on support for the native pipe?

**Comment by nischalshrestha** [comment]
> Any update on support for the native pipe?

Hi @andreweatherman! I'll be doing some cleanup on the code splitting and analysis part soon, it is a bit messy and hard to maintain. During this process, I'll figure out how to add support for the native pipe.

**Comment by andreweatherman** [comment]
> > Any update on support for the native pipe?
> 
> 
> 
> Hi @andreweatherman! I'll be doing some cleanup on the code splitting and analysis part soon, it is a bit messy and hard to maintain. During this process, I'll figure out how to add support for the native pipe.

I forked the repo and added a bit of stuff to make the native pipe work; but it only works with the Shiny viewer right now (so no intermediate pipe steps, etc.).

**Comment by nischalshrestha** [comment]
> > > Any update on support for the native pipe?
> > 
> > 
> > Hi @andreweatherman! I'll be doing some cleanup on the code splitting and analysis part soon, it is a bit messy and hard to maintain. During this process, I'll figure out how to add support for the native pipe.
> 
> I forked the repo and added a bit of stuff to make the native pipe work; but it only works with the Shiny viewer right now (so no intermediate pipe steps, etc.).

Nice! I think the approach of substituting the base for the magrittr pipe is fine in the code text itself works. But, I'm a bit wary on whether the behavior is exactly the same btw `%>%` and `|>` pipes. Think for tidyverse code should work fine, and only issue I can think of rn is if you base piped into other things besides `dplyr`/`tidyr` like lambdas/ggplot2 functions etc. 🤔

**Comment by andreweatherman** [comment]
> > > > Any update on support for the native pipe?
> 
> > > 
> 
> > > 
> 
> > > Hi @andreweatherman! I'll be doing some cleanup on the code splitting and analysis part soon, it is a bit messy and hard to maintain. During this process, I'll figure out how to add support for the native pipe.
> 
> > 
> 
> > I forked the repo and added a bit of stuff to make the native pipe work; but it only works with the Shiny viewer right now (so no intermediate pipe steps, etc.).
> 
> 
> 
> Nice! I think the approach of substituting the base for the magrittr pipe in the code text itself works. But, I'm a bit wary on whether the behavior is exactly the same btw `%>%` and `|>` pipes. Think for tidyverse code should work fine, and only issue I can think of rn is if you base piped into other things besides `dplyr`/`tidyr` like lambdas/ggplot2 functions etc. 🤔 
> 
> 
> 
> 
> 
> 
> 
> 

Yeah, the small functionality differences could cause some problems, but those changes were purely to add support for some basic tidyverse stuff (dplyr, tidyr, etc.). Looking forward to seeing what you come up with for a full extension!
